db/index.ts

```diff
@@ -28,23 +29,81 @@ CREATE TABLE IF NOT EXISTS settings (
 );
 `;
 
-export const expo = SQLite.openDatabaseSync('period.db');
+let expo: SQLite.SQLiteDatabase | null = null;
+let db: ReturnType<typeof drizzle> | null = null;
 
-// Execute the migration
-expo.execSync(MIGRATION_TABLES);
+export async function deleteDatabaseFile(): Promise<void> {
+  if (expo) {
+    await expo.closeAsync();
+  }
+  
+  expo = null;
+  db = null;
+  
+  await SQLite.deleteDatabaseAsync('period.db');
+}
+
+export async function initializeDatabase(): Promise<void> {
+  if (db) {
+    return;
+  }
+
+  try {
+    await initializeEncryption();
+    
+    const key = getEncryptionKey();
+    const hexKey = Array.from(key).map(b => b.toString(16).padStart(2, '0')).join('');
```

Comment:
I didn't realise that SQLite required the key to be hex-encoded. I thought it could be binary. Seeing that, I'd suggest replacing base64 encoding with hex encoding in `databaseEncryptionService.ts`. Whilst you're there, also rename `getEncryptionKey()` to `getEncryptionKeyHex()` to be explicit.